_posts/2017-12-11-away-week-prague/index.md

```diff
@@ -0,0 +1,62 @@
+
+![team-photo][team]
+
+<h2 id="hackathon">Hackathon</h2>
+
+Mid-week, we undertook the traditional Snowplow Away Week Hackathon: an entire day of meticulous planning, programming, strategizing, and building around some of Snowplow’s most unique and under-investigated challenges and questions on how our products work. Five teams spent their time tackling knowledge sharing, company messaging, web development, internal data collection, and data modeling. As the teams presented the fruits of their labors over pizza that evening, it was obvious that these ideas, despite being in their early stages, would be significant improvements for Snowplow and there have already been discussions about various implementations.
+
+One core reason we’re able to be a valuable resource for our users is our understanding of the challenges they’re facing, and we’ve taken steps to more comprehensively understand the process and benefits of A/B testing. Knowing that A/B testing is a high priority for many of our users, and having come across positive feedback for [Wasabi][wasabi], an open source testing platform by Intuit, one of our hackathon teams focused on setting up and integrating a Wasabi instance, allowing us to configure a wide range of experiments for our website and products. Given the success of this test, we’re even exploring offering Wasabi as a tool for our users, letting them quickly and seamlessly test hypotheses about their website or product and collect data with their Snowplow pipeline.
+
+The Snowplow user experience was top of mind for our hackathon teams. Beyond A/B testing, we also had a team focused on data modeling. The data modeling step of the digital analytic process can often be the most challenging, with access to so much data often making it difficult to know what questions to ask to get started. This team explored ways we can improve the data modeling process for our users, specifically focusing on three different areas: building a DSL, porting data to graph databases using [Neo4j][neo], and making the modeling process more accessible with PySpark. You can look forward to further information about a more streamlined, user friendly data modeling process in a future post. To further empower our users, another team set about creating what they called Snowplow Wire. Using Wire, we turned our analytic capabilities inward on our pipeline, allowing us to collect brand new data on the health and performance of the pipelines we manage as well as how to socialize this metadata to our users so they can benefit from it as well.
+
```

Comment:
Let's add here that data modeling is a key step in the Snowplow data pipeline: it is where "meaning" is effectively layered onto the underlying event data and that data then becomes easier to socialize around the business. Given how critical this step is, it is a real pain point today that the data modeling process effectively lives in SQL. In the hackathon, we attempted to develop three alternatives to the SQL based approach: (i) developing a data modeling DSL from scratch, (ii) building a data modeling process in a graph database using neo4j and (iii) building a data modeling process in PySpark.